### PR TITLE
Fix bugs 

### PR DESCRIPTION
**Pull request by JerryM-Quectel**
These bugs found when the helper called by modemfwd

### CODE REVIEW

File: ql-mbim-core.c [hunk 1]
```diff
@@ -46,9 +46,9 @@ static int log_printf(int lvl, const char *log_msg)
     } while (0)
 
 static int file_get_value(const char *fpath, int base);
-static int flash_mode_check(void);
+int flash_mode_check(void);
```
Comment (naveenquectel) [inline ql-mbim-core.c:50]: flash_mode_check option is not required by modemfwdupd for now and this function can remain static.

File: ql-mbim-core.c [hunk 2]
```diff
@@ -614,10 +614,10 @@ int mbim_reboot_modem(void)
          * 2. call quectel-modemfwd-helper --reboot
          *
          * In fact, when step1 finished, the module reboot itself automatically.
-         * Return 1 to the caller pretending the call is successful.
+         * Return success value to the caller pretending the call is successful.
          */
         info_printf("Could not find a Quectel modem available for commands!\n");
-        return 1;
+        return 0;
```
Comment (naveenquectel) [inline ql-mbim-core.c:620]: success and non-success values can use macros to maintain consistency across the code. This needs to be modified by defining below two macros in some common header files. 
#define _EXIT_SUCCESS 0
#define _EXIT_FAILURE -1

File: ql-modem-helper.c [hunk 3]
```diff
@@ -107,13 +111,15 @@ int main(int argc, char *argv[])
             {"get_fw_info", 0, NULL, 'G'},
             {"prepare_to_flash", 0, NULL, 'P'},
             {"flash_fw", 1, NULL, 'A'},
+            {"flash_mode_check", 0, NULL, 'M'},
             {"reboot", 0, NULL, 'R'},
-            {"sahara-reboot", 0, NULL, 'S'},
+            {"clear_attach_apn", 0, NULL, 'C'},
```
Comment (naveenquectel) [inline ql-modem-helper.c:116]: Please add this option when we add the functionality to this option, currently this option is not doing any meaningful logic.

File: ql-sahara-core.c [hunk 4]
```diff
@@ -783,168 +793,4 @@ int sahara_flash_all(char *main_file_path,char * oem_file_path,char* carrier_fil
     }
     qdl_close(&qdl);
     return 0;
-}
```
Comment (naveenquectel) [inline ql-sahara-core.c:786]: Do not make huge changes in one single commit, just do only one logical change in one bug-fix commit.

File: ql-mbim-core.c
Comment (naveenquectel) [inline ql-mbim-core.c:1]: These changes can not be merged for below reasons. 

1. Changes in the current commits handles many different logical changes and code improved and need to applied in multiple commits for readability. 
2. Function return for success and failure case are not consistent and improvement should be made to have consistent return status from the global functions. 
3. New cases added in the main function should have logical change defined.

**Review by naveenquectel — CHANGES_REQUESTED** [review]

**Review by naveenquectel — CHANGES_REQUESTED** [review]
Function return for success and failure case are not consistent and improvement should be made to have consistent return status from the global functions.

**Review by naveenquectel — CHANGES_REQUESTED** [review]
Why we need 5 second delay? Delays in the codes are not required if protocol are specification does not define the same.

**Review by naveenquectel — APPROVED** [review]
This change looks good to me.

**Review by naveenquectel — COMMENTED** [review]
This change need to be verified before merge.

File: ql-modem-helper.c (hunk 3 above)
Comment (JerryM-Quectel) [reply ql-modem-helper.c:116]: > Please add this option when we add the functionality to this option, currently this option is not doing any meaningful logic.

modemfwd would flash carrier fw again if --clear_attach_apn return failure.

**Comment by JerryM-Quectel** [comment]
> Why we need 5 second delay? Delays in the codes are not required if protocol are specification does not define the same.

Sometimes after --prepare_to_flash  exit,  module switching to download mode is still in progress.
If in this time modemfwd call   "qmodemhelper --flash_fw ...",  would fail.

so before --prepare_to_flash exit, make sure module alread in download mode.

**Pull request merged**